argparse.cc

```diff
@@ -0,0 +1,108 @@
+#include "src/node_options.h"
+#include "src/util.h"
+
+#include <iostream>
+#include <ranges>
+#include <string>
+#include <vector>
+
+extern "C" {
+
+uint64_t uv_get_total_memory() {
+  return 0;
+}
+
+uint64_t uv_get_constrained_memory() {
+  return 0;
+}
+};
+
+namespace node {
+
+void Assert(const AssertionInfo& info) {
+  fprintf(stderr,
+          "\n"
+          "  #  %s at %s\n"
+          "  #  Assertion failed: %s\n\n",
+          info.function ? info.function : "(unknown function)",
+          info.file_line ? info.file_line : "(unknown source location)",
+          info.message);
+
+  fflush(stderr);
+
+  abort();
+}
+
+}  // namespace node
+
+int main(int argc, char** argv) {
+  std::vector<std::string> args(argv, argv + argc);
+  std::vector<std::string> exec_args;
+  std::vector<std::string> errors;
+
+  node::PerProcessOptions cli_options;
+
+  cli_options.cmdline = args;
+
+  if (const char* result = std::getenv("NODE_OPTIONS")) {
+    std::string node_options(result);
+
+    std::vector<std::string> env_argv =
+        node::ParseNodeOptionsEnvVar(node_options, &errors);
+    for (auto error : errors) {
+      std::cout << "error: " << error << std::endl;
+    }
+    if (!errors.empty()) return 1;
+
+    env_argv.insert(env_argv.begin(), args.at(0));
+
+    {
+      std::vector<std::string> v8_args;
+      node::options_parser::Parse(&env_argv,
+                                  nullptr,
+                                  &v8_args,
+                                  &cli_options,
+                                  node::OptionEnvvarSettings::kAllowedInEnvvar,
+                                  &errors);
+
+      for (auto arg : v8_args | std::views::drop(1)) {
+        std::cout << "v8_arg: " << arg << std::endl;
+      }
+    }
+  }
+
+  node::HandleEnvOptions(cli_options.per_isolate->per_env,
```

Comment:
How would this be maintained going forward? For example, this creates a dependence on the current shape of `PerProcessOptions` etc, which are very much just ad-hoc internal data structures that are very subject to change. Things like `HandleEnvOptions` or `Parse` are also subject to internal refactoring, like for example adding a different mechanism than `HandleEnvOptions` for creating env var/flag pairs like `EnvImplies("NODE_PENDING_DEPRECATION", "--pending-deprecation")` to reduce the use of lambdas then this would be ditched.

Reply:
If compatibility is not a concern, then I think it makes more sense to just do a CI elsewhere instead of checking the test into the Node.js source, because then the maintainability burden would be shifted to Node.js contributors, and may create barriers to new contributors if they don't know the context and somehow think that breaking this test is bad. We can keep the macros internally to reduce the merge conflicts, but keeping the test in the CI doesn't seem very helpful if it's supposed to be maintained by another project.

Reply:
I'm think it is not guaranteed that node_options.h/cc has no dependencies like uv/libnode/v8/etc.

Reply:
Yeah I feel that to address https://github.com/nodejs/node/issues/46339 we might end up adding a v8 dependency to the option parser....